components/concept/toolbar/src/main/java/mozilla/components/concept/toolbar/Toolbar.kt

```diff
@@ -356,4 +361,22 @@ interface Toolbar {
         INSECURE,
         SECURE,
     }
+
+    /**
+     * Indicates which tracking protection status a site has.
+     */
+    enum class SiteTrackingProtection {
+        /**
+         * The site has tracking protection enabled, but none trackers have been blocked or detected.
+         */
+        ON,
+        /**
+         * The site has tracking protection enabled, and trackers have been blocked or detected.
+         */
+        ACTIVE,
```

Comment:
Are those the same name of states we'd get from e.g. GV? It's hard to distinguish ON and ACTIVE just from the names. For me "ACTIVE" could also mean "enabled but no trackers blocked". Can we find better names? I wouldn't even be opposed to very descriptive names like ON_TRACKERS_BLOCKED, ON_NO_TRACKERS. wdyt?

Reply:
> Are those the same name of states we'd get from e.g. GV?

No GV, doesn't do similar states, this is something that UX/UI came up, for users easily understand what tracking protection is doing for them.

> Could also mean "enabled but no trackers blocked". Can we find better names? I wouldn't even be opposed to very descriptive names like ON_TRACKERS_BLOCKED, ON_NO_TRACKERS.

I like the names 👍

Reply:
The mocks above also show an "OPENED" state - do we need to support that here too?

Reply:
@pocmo  do we want to do the same with the drawable properties?  We could end up with massive variables!

```kotlin
var trackingProtectionONIcon: Drawable
        get() = displayToolbar.trackingProtectionONIcon
        set(value) {
            displayToolbar.trackingProtectionONIcon = value
        }

    /**
     * Sets the icon of the tracking protection icon when is active
     * e.g (Trackers have been blocked or detected)
     */
    var trackingProtectionActiveIcon: Drawable
        get() = displayToolbar.trackingProtectionActiveIcon
        set(value) {
            displayToolbar.trackingProtectionActiveIcon = value
        }
```

Reply:
`The mocks above also show an "OPENED" state - do we need to support that here too?`

Yeah, we told with UX and we agree to don't do it 🚫